relay-cardinality/src/limiter.rs

```diff
@@ -165,19 +166,24 @@ impl<T: Limiter> CardinalityLimiter<T> {
         }
 
         metric!(timer(CardinalityLimiterTimers::CardinalityLimiter), {
-            let entries = items.iter().enumerate().filter_map(|(id, item)| {
-                Some(Entry::new(
-                    EntryId(id),
-                    item.namespace()?,
-                    item.name(),
-                    item.to_hash(),
-                ))
-            });
+            let entries = items
+                .iter()
+                .enumerate()
+                .filter_map(|(id, item)| {
+                    Some(Entry::new(
+                        EntryId(id),
+                        item.namespace()?,
+                        item.name(),
+                        item.to_hash(),
+                    ))
+                })
+                .collect::<Vec<_>>();
```

Comment:
This is unfortunate, the code before was explicitly built to borrow and not need to allocate here.